### PR TITLE
Corrects token null clauses and checks

### PR DESCRIPTION
**Pull request by MrCsabaToth**
Also declares some vars as finals

### CODE REVIEW

File: lib/API/Oauth.dart [hunk 1]
```diff
@@ -233,13 +207,13 @@ abstract class Auth {
 
     // Check if the token is not expired
     // if (_token != null) {
-    if ((_token != null) || (_token == "null")) {
+    if (_token != null && _token != "null") {
```
Comment (MrCsabaToth) [inline lib/API/Oauth.dart:210]: I think this should be a logical AND operator, and we'd need to flip the second clause? We supposed to test here if the token is not null?

Comment (MrCsabaToth) [reply lib/API/Oauth.dart:210]: Also removing some unnecessaary parantheses

File: lib/API/Oauth.dart [hunk 2]
```diff
@@ -233,13 +207,13 @@ abstract class Auth {
...
       globals.displayInfo(
           'token has been stored before! ${tokenStored.accessToken}  exp. ${tokenStored.expiresAt}');
     }
 
     // Use the refresh token to get a new access token
-    if (isExpired && ((_token != "null") || (_token != null))) {
+    if (isExpired && _token != null && _token != "null") {
```
Comment (MrCsabaToth) [inline lib/API/Oauth.dart:216]: Here if either token is null or is "null" then the right hand side of the logical and becomes true. I feel like that the logical OR should be an AND because we want to test for the condition of the token is not null (which can have two variants). Bunch of extra parantheses removed.

File: lib/API/Oauth.dart [hunk 3]
```diff
@@ -253,9 +227,7 @@ abstract class Auth {
     }
 
     // Check if the scope has changed
-    if ((tokenStored.scope != scope) ||
-        (_token == "null") ||
-        (_token == null)) {
+    if (tokenStored.scope != scope || _token == "null" || _token == null) {
```
Comment (MrCsabaToth) [inline lib/API/Oauth.dart:230]: I only removed parantheses here.

File: lib/API/Oauth.dart (hunk 1 above)
Comment (BirdyF) [reply lib/API/Oauth.dart:210]: You are right, it should be a logical AND.
But I am not sure that "null" is a value that is returned.

Comment (MrCsabaToth) [reply lib/API/Oauth.dart:210]: Yes, neither do I know if "null" can actually be a valid value, but the code can prepare for it. The change below is another spot where "null" is involved. I still haven't tested this fully, I hope the merge won't screw up anything.
